src/lcm/dispatchers.py

```diff
@@ -1,13 +1,13 @@
 import inspect
 from collections.abc import Callable
-from typing import Literal, TypeVar
+from typing import Literal, TypeVar, cast
 
 from jax import Array, vmap
 
 from lcm.functools import allow_args, allow_only_kwargs
 from lcm.utils import find_duplicates
 
-F = TypeVar("F", bound=Callable[..., Array])
```

Comment:
In functools, you are using `ReturnType` for such a construct. I think I prefer the more verbose version.

Reply:
Here the return type is known to be Array or tuple[Array, Array]. The construct `F` would map to the construct `Callable[..., ReturnType]` in functools.

I think there is room for improvement in the typing, but this would require me to look up advanced typing features. FYI, we are doing slightly more than JAX is doing in their code-base (see [here](https://github.com/jax-ml/jax/blob/44872296a139204ef9d4ca102859deb7305f8707/jax/_src/api.py#L94)).

Reply:
I just meant the string "F" (one character, could be anything) vs "ReturnType" (10 characters, allows to guess what is meant)!

Reply:
Personally, I like `F`, as it is also what the JAX source-code is using for `vmap`. If you want something more declarative, I'd propose `FunctionWithArrayReturn`?